overlord/snapstate/handlers.go

```diff
@@ -1906,7 +1906,9 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 		}
 	}
 	if !rebootInfo.RebootRequired || canReboot {
-		m.maybeRestart(t, newInfo, rebootInfo)
+		if err := m.finishTaskWithMaybeRestart(t, finalStatus, newInfo, rebootInfo); err != nil {
+			return err
+		}
 	}
```

Comment:
could this be
```suggestion
	} else {
		t.setStatus(finalStatus)
	}
```
and remove the call to `t.setStaus()` above, like you did in other handlers?

Reply:
we can do it here I think